install_ffmpeg.sh

```diff
@@ -15,24 +42,34 @@ if [ ! -e "$HOME/nasm/nasm" ]; then
   make install || echo "Installing docs fails but should be OK otherwise"
 fi
 
-if [ ! -e "$HOME/x264/x264" ]; then
+if [ ! -e "$HOME/x264" ]; then
   git clone http://git.videolan.org/git/x264.git "$HOME/x264"
   cd "$HOME/x264"
   # git master as of this writing
   git checkout 545de2ffec6ae9a80738de1b2c8cf820249a2530
-  ./configure --prefix="$HOME/compiled" --enable-pic --enable-static
+  ./configure --prefix="$HOME/compiled" --enable-pic --enable-static ${HOST_OS:-} --disable-cli
   make
   make install-lib-static
 fi
 
+EXTRA_FFMPEG_FLAGS=""
+# Only Linux supports CUDA... for now.
+if [ $(uname) == "Linux" ]; then
+  if [ -e /usr/local/cuda/include ]; then
+    echo "CUDA detected, building with GPU support"
+    EXTRA_FFMPEG_FLAGS="--enable-cuda --enable-cuvid --enable-nvenc --enable-filter=scale_cuda --enable-encoder=h264_nvenc --enable-cuda-nvcc"
+  fi
+fi
+
 if [ ! -e "$HOME/ffmpeg/libavcodec/libavcodec.a" ]; then
-  git clone -b n4.1 https://git.ffmpeg.org/ffmpeg.git "$HOME/ffmpeg" || echo "FFmpeg dir already exists"
+  git clone https://git.ffmpeg.org/ffmpeg.git "$HOME/ffmpeg" || echo "FFmpeg dir already exists"
   cd "$HOME/ffmpeg"
-  ./configure --disable-programs --disable-doc --disable-sdl2 --disable-iconv \
+  git checkout 4cfc34d9a8bffe4a1dd53187a3e0f25f34023a09
+  ./configure ${TARGET_OS:-} --disable-programs --disable-doc --disable-sdl2 --disable-iconv \
     --disable-muxers --disable-demuxers --disable-parsers --disable-protocols \
     --disable-encoders --disable-decoders --disable-filters --disable-bsfs \
     --disable-postproc --disable-lzma \
-    --enable-gnutls --enable-libx264 --enable-gpl \
+    --enable-gnutls --enable-libx264 --enable-gpl --enable-version3 --enable-nonfree \
```

Comment:
Curious, what brought in the GPLv3 dependency?

Reply:
Hmm, apparently nothing. Must have copied it there accidentally when looking up how to spell `--enable-nonfree`, good catch